crates/turborepo-lib/src/commands/mod.rs

```diff
@@ -81,62 +88,41 @@ impl CommandBase {
         Ok(())
     }
 
-    fn create_user_config(&self) -> Result<()> {
-        let user_config = UserConfigLoader::new(default_user_config_path()?)
+    fn user_config_init(&self) -> Result<UserConfig, ConfigError> {
+        UserConfigLoader::new(default_user_config_path()?)
             .with_token(self.args.token.clone())
-            .load()?;
-        self.user_config.set(user_config)?;
-
-        Ok(())
-    }
-
-    fn create_client_config(&self) -> Result<()> {
-        let client_config = ClientConfigLoader::new()
-            .with_remote_cache_timeout(self.args.remote_cache_timeout)
-            .load()?;
-        self.client_config.set(client_config)?;
-
-        Ok(())
+            .load()
     }
 
-    pub fn repo_config_mut(&mut self) -> Result<&mut RepoConfig> {
-        if self.repo_config.get().is_none() {
-            self.create_repo_config()?;
-        }
-
-        Ok(self.repo_config.get_mut().unwrap())
-    }
-
-    pub fn repo_config(&self) -> Result<&RepoConfig> {
-        if self.repo_config.get().is_none() {
-            self.create_repo_config()?;
-        }
-
-        Ok(self.repo_config.get().unwrap())
+    pub fn user_config(&self) -> Result<&UserConfig, ConfigError> {
+        self.user_config.get_or_try_init(|| self.user_config_init())
     }
 
-    pub fn user_config_mut(&mut self) -> Result<&mut UserConfig> {
-        if self.user_config.get().is_none() {
-            self.create_user_config()?;
-        }
+    pub fn user_config_mut(&mut self) -> Result<&mut UserConfig, ConfigError> {
+        // Approximates `get_mut_or_try_init`
+        self.user_config
+            .get_or_try_init(|| self.user_config_init())?;
```

Comment:
Same comment as above.
```suggestion
        self.user_config()?;
```